crates/ruff_python_parser/src/semantic_errors.rs

```diff
@@ -125,6 +128,37 @@ impl SemanticSyntaxChecker {
                 Self::invalid_star_expression(target, ctx);
                 Self::invalid_star_expression(iter, ctx);
             }
+            Stmt::Try(ast::StmtTry {
+                range: _,
+                body,
+                handlers,
+                orelse,
+                finalbody,
+                is_star: _,
+            }) => {
+                let mut visitor = TryExceptVisitor::new(ctx);
+                for stmt in body {
+                    SourceOrderVisitor::visit_stmt(&mut visitor, stmt);
+                }
```

Comment:
We should use the `visit_body` method instead in here, orelse, handlers, and finalbody.

Reply:
except for the "handlers" loop

Reply:
I think the easiest here is to call `walk_stmt(&visitor, stmt)` because this is an exact reimplementation of visiting a try statement.

Reply:
You're right, we traverse into function statements but we increase the `scope` counter. I have to take another look to understand what the scope counter is even used for but I suspect that different rules apply for names and global statements inside nested functions, classes and it's unclear to me if that also applies to lambdas (can't have any global statements but it may have identifiers)